src/wazuh_modules/task_manager/wm_task_manager_db.c

```diff
@@ -92,12 +116,134 @@ int wm_task_manager_check_db() {
     return 0;
 }
 
+void* wm_task_manager_clean_db(void *arg) {
+    wm_task_manager *config = (wm_task_manager *)arg;
+    time_t next_clean = time(0);
+    time_t next_timeout = next_clean;
+
+    while (1) {
+        time_t now = time(0);
+        time_t sleep_time = 0;
+
+        if (now >= next_timeout) {
+            // Set the status of old tasks IN PROGRESS to TIMEOUT
+            next_timeout = now + WM_TASK_MAX_IN_PROGRESS_TIME;
+            wm_task_manager_set_timeout_status(now, &next_timeout);
+        }
+
+        if (now >= next_clean) {
+            // Delete entries older than cleanup_time
+            next_clean = now + WM_TASK_CLEANUP_DB_SLEEP_TIME;
+            wm_task_manager_delete_old_entries((now - config->cleanup_time));
+        }
+
+        if (next_timeout < next_clean) {
+            sleep_time = next_timeout - now;
```

Comment:
Suggestion: take a look at `w_sleep_until` method. It sleeps one second at a time which helps to avoid problems when the user changes the clock time while it is waiting. It is not completely necessary in this point though so I will leave it up to you

Reply:
I changed it.